File: app/src/main/java/peterkotik/com/emojichat/screens/camerapreview/CameraPreviewContract.java
package peterkotik.com.emojichat.screens.camerapreview;

import com.affectiva.android.affdex.sdk.detector.Face;

import java.util.List;

interface CameraPreviewContract {
    interface View {
        void setEmojiButtonText(String text);

        void setEmojiButtonVisible(boolean visible);

    }

    interface Presenter {
        boolean onImageResult(List<Face> faces);

    }
}

File: app/src/main/java/peterkotik/com/emojichat/screens/camerapreview/CameraPreviewContract.kt
package peterkotik.com.emojichat.screens.camerapreview

import com.affectiva.android.affdex.sdk.detector.Face

internal interface CameraPreviewContract {
    interface View {
        fun setEmojiButtonText(text: String)

        fun setEmojiButtonVisible(visible: Boolean)

    }

    interface Presenter {
        fun onImageResult(faces: List<Face>): Boolean

    }
}
